lib/handler/redirect.c

```diff
@@ -79,10 +85,43 @@ static void redirect_internally(h2o_redirect_handler_t *self, h2o_req_t *req, h2
     h2o_send_error_503(req, "Internal Server Error", "internal server error", 0);
 }
 
+static h2o_iovec_t join_list(h2o_mem_pool_t *pool, h2o_iovec_vector_t *list, h2o_iovec_t delimiter)
+{
+    assert(list->size > 0);
+    size_t joined_len = 0;
+    h2o_iovec_t *joined = alloca(sizeof(*joined) * (list->size * 2 - 1));
+
+    size_t i;
+    for (i = 0; i != list->size; ++i) {
+        if (i != 0) {
+            joined[joined_len++] = delimiter;
+        }
+        joined[joined_len++] = list->entries[i];
+    }
+    return h2o_concat_list(pool, joined, joined_len);
+}
+
+static void build_prefix_list(const char *str, h2o_iovec_vector_t *dest)
+{
+    *dest = (h2o_iovec_vector_t){ 0 };
+    char *p = (char *)str;
+    char *found;
```

Comment:
`p` and `found` can be `const char *`, as `strchr` is a function that takes `const char *` as an argument and returns `char *`.